nix/jjui.nix

```diff
@@ -0,0 +1,78 @@
+{ lib, ... }:
+{
+  perSystem =
+    { pkgs, ... }:
+    let
+      buildInputs = [ pkgs.jujutsu ]; # Runtime
+      nativeBuildInputs = [ pkgs.makeWrapper ]; # Buildtime
+
+      jjui = pkgs.buildGoModule {
+        inherit buildInputs nativeBuildInputs;
+
+        pname = "jjui";
+        version = "dev";
+
+        src = lib.fileset.toSource {
+          root = ./..;
+          fileset = lib.fileset.unions [
+            ./../go.mod
+            ./../go.sum
+            ./../cmd
+            ./../internal
+            ./../test
+          ];
+        };
+        vendorHash = lib.strings.trim (builtins.readFile ./vendor-hash);
+        doCheck = false;
+
+        postInstall = ''
+          wrapProgram $out/bin/jjui \
+            --prefix PATH : ${pkgs.lib.makeBinPath [ pkgs.jujutsu ]}
```

Comment:
I'm a bit worried about this, because this is is making the result binary use a fixed jujutsu version. The one from nixpkgs. Most of the time this will work, but many people (like me) would at times need to use more-recent versions of jujutsu than that currently at nixpkgs.

The way this is commonly solved is to have a nix file for the package and use pkgs.callPackage on it, so people is able to override the jujutsu package they want to use.

Reply:
The idea behind this is to have a pinned `jj` version to make sure `jjui` supports it. True, they could use an overlay but `pkgs.callPackage` seems like the defacto way.

Reply:
I think it is best we not to wrap jjui, I know it is a runtime dependency, wrapping would make any other jj on PATH just being ignored since the package will always use the one that comes with nixpkgs. What do you think? Nix people can easily wrap the result of this flake if they needed.

Reply:
I got used to doing this for packages such as `lazygit` to ensure compatability but it seems that in jjui's docs:
Minimum supported `jj` version is `v0.26+` and this flake is using `v0.36.0` so it should work just fine. Agreed, Nix users can wrap it themselves, but in this case I would leave `jujutsu` in `buildInputs` so users have `jj` when using `jjui`. Thoughts?

Reply:
~Having only buildInputs is fine I guess.~ lets get rid of the wrappers.

Hm, buildInputs is for runtime-dependencies for installStep and friends? our current nix code has no buildInputs and everything has been fine. I'm now wondering if we even need buildInputs.